File: examples/EvalDemo.swift
import Foundation

@main
struct EvalDemoMain {
    static func main() throws {
        let root: URL
        if CommandLine.arguments.count > 1 {
            root = URL(fileURLWithPath: CommandLine.arguments[1])
        } else {
            root = URL(fileURLWithPath: FileManager.default.currentDirectoryPath)
        }
        let manifestURL = root.appendingPathComponent("schemas/demo.manifest.json")
        let data = try Data(contentsOf: manifestURL)
        let manifest = try JSONDecoder().decode(FlagsManifest.self, from: data)
        let ev = Evaluator(manifest: manifest)
        let res = ev.evaluate("checkout.promo-banner", EvalContext(userId: "user-alice", env: "staging"))
        if !res.found || res.source.rawValue.isEmpty {
            fputs("empty result\n", stderr)
            exit(1)
        }
        print("key=\(res.key) value=\(res.value.debugText) source=\(res.source.rawValue)")
    }
}

File: examples/eval-demo.mjs
import { readFileSync } from "node:fs";
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";
import { Evaluator } from "../ts/evaluator.ts";

const root = join(dirname(fileURLToPath(import.meta.url)), "..");
const manifest = JSON.parse(
  readFileSync(join(root, "schemas", "demo.manifest.json"), "utf8"),
);
const ev = new Evaluator(manifest);
const res = ev.evaluate("checkout.promo-banner", {
  userId: "user-alice",
  env: "staging",
});
if (!res.found || res.value === undefined || res.value === null || !res.source) {
  console.error("empty result", res);
  process.exit(1);
}
console.log(`key=${res.key} value=${JSON.stringify(res.value)} source=${res.source}`);
